### ISSUE TITLE
[Dreamcast] unordered_map::at after sending 99 00 04 00 when moving to lobby

### ISSUE THREAD

**Issue by Matt-Swift** [fuzziqersoftware/newserv#466]
**Describe the bug**
When a player attempts to go to lobby, they immediately receive Error 100. The server logs show an `unordered_map::at` error after the `99 00 04 00` packet.

However, nobody else hits this error, it is *only* this one player, which is a little strange. This happens on both JPv2 and EUv2.

**To reproduce**
1. Player goes to lobby after connecting successfully

**Game version(s) (choose one or more of the following):**
DC v2

**Server log output**
```
newserv> I 229195 2024-04-19 08:48:33 - [C-1] Created
I 229195 2024-04-19 08:48:33 - [Server] Client connected: C-1 on fd 55 via 19 (TG-9200-GC_V3-gc-eu10-login_server)
I 229195 2024-04-19 08:48:33 - [Commands] Sending to C-1 (version=GC_V3 command=17 flag=00)
0000 | 17 00 00 01 44 72 65 61 6D 43 61 73 74 20 50 6F |     DreamCast Po
0010 | 72 74 20 4D 61 70 2E 20 43 6F 70 79 72 69 67 68 | rt Map. Copyrigh
0020 | 74 20 53 45 47 41 20 45 6E 74 65 72 70 72 69 73 | t SEGA Enterpris
0030 | 65 73 2E 20 31 39 39 39 00 00 00 00 00 00 00 00 | es. 1999        
0040 | 00 00 00 00 76 8C 0E 36 47 11 CD DD 54 68 69 73 |     v  6G   This
0050 | 20 73 65 72 76 65 72 20 69 73 20 69 6E 20 6E 6F |  server is in no
0060 | 20 77 61 79 20 61 66 66 69 6C 69 61 74 65 64 2C |  way affiliated,
0070 | 20 73 70 6F 6E 73 6F 72 65 64 2C 20 6F 72 20 73 |  sponsored, or s
0080 | 75 70 70 6F 72 74 65 64 20 62 79 20 53 45 47 41 | upported by SEGA
0090 | 20 45 6E 74 65 72 70 72 69 73 65 73 20 6F 72 20 |  Enterprises or 
00A0 | 53 4F 4E 49 43 54 45 41 4D 2E 20 54 68 65 20 70 | SONICTEAM. The p
00B0 | 72 65 63 65 64 69 6E 67 20 6D 65 73 73 61 67 65 | receding message
00C0 | 20 65 78 69 73 74 73 20 6F 6E 6C 79 20 74 6F 20 |  exists only to 
00D0 | 72 65 6D 61 69 6E 20 63 6F 6D 70 61 74 69 62 6C | remain compatibl
00E0 | 65 20 77 69 74 68 20 70 72 6F 67 72 61 6D 73 20 | e with programs 
00F0 | 74 68 61 74 20 65 78 70 65 63 74 20 69 74 2E 00 | that expect it. 
I 229195 2024-04-19 08:48:33 - [Commands] Received from C-1 (version=GC_V3 command=9A flag=00)
0000 | 9A 00 E0 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0010 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0020 | 00 00 00 00 37 33 34 39 38 32 44 34 00 00 00 00 |     734982D4    
0030 | 00 00 00 00 FF FF FF FF FF FF FF FF 00 00 00 00 |     aaaaaaaa    
0040 | 00 00 00 00 00 00 80 64 26 5F 67 F7 25 00 00 00 |        d&_g %   
0050 | 6C 75 6B 61 00 00 00 00 00 00 00 00 00 00 00 00 | luka            
0060 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0070 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0080 | 6C 75 6B 61 6B 6F 00 00 00 00 00 00 00 00 00 00 | lukako          
0090 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00A0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00B0 | 6C 75 6B 61 6B 6F 40 63 75 6D 2E 74 6F 77 6E 00 | lukako@cum.town
00C0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00D0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
I 229195 2024-04-19 08:48:33 - [C-1] Game version changed to DC_V2
I 229195 2024-04-19 08:48:33 - [Commands] Sending to C-1 (version=DC_V2 command=9A flag=02)
0000 | 9A 02 04 00                                     |                 
I 229195 2024-04-19 08:48:33 - [Commands] Received from C-1 (version=DC_V2 command=9D flag=00)
0000 | 9D 00 30 01 00 00 FF FF FF FF 00 00 00 00 80 64 |   0            d
0010 | 26 5F 67 F7 25 00 00 00 00 01 00 00 00 00 00 00 | &_g %           
0020 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0030 | 00 00 00 00 00 00 00 00 00 00 00 00 37 33 34 39 |             7349
0040 | 38 32 44 34 00 00 00 00 00 00 00 00 FF FF FF FF | 82D4        aaaa
0050 | FF FF FF FF 00 00 00 00 00 00 00 00 6C 75 6B 61 | aaaa        luka
0060 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0070 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0080 | 00 00 00 00 00 00 00 00 00 00 00 00 6C 75 6B 61 |             luka
0090 | 6B 6F 00 00 00 00 00 00 00 00 00 00 00 00 00 00 | ko              
00A0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00B0 | 00 00 00 00 00 00 00 00 00 00 00 00 4C 75 6B 61 |             Luka
00C0 | 6B 6F 00 00 00 00 00 00 00 00 00 00 00 00 00 00 | ko              
00D0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00E0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
00F0 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0100 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0110 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
0120 | 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 |                 
I 229195 2024-04-19 08:48:33 - [C-1] Game version changed to DC_V2
I 229195 2024-04-19 08:48:33 - [Commands] Sending to C-1 (version=DC_V2 command=04 flag=00)
0000 | 04 00 0C 00 00 00 01 00 D4 82 49 73             |           Is    
I 229195 2024-04-19 08:48:33 - [Commands] Sending to C-1 (version=DC_V2 command=07 flag=04)
0000 | 07 04 90 00 11 00 00 11 FF FF FF FF 04 00 45 55 |               EU
0010 | 2F 52 61 67 6F 6C 00 00 00 00 00 00 00 00 00 00 | /Ragol          
0020 | 11 00 00 11 11 22 22 11 04 0F 47 6F 20 74 6F 20 |      ""   Go to 
0030 | 6C 6F 62 62 79 00 00 00 00 00 00 00 11 00 00 11 | lobby           
0040 | 11 44 44 11 04 0F 44 6F 77 6E 6C 6F 61 64 20 71 |  DD   Download q
0050 | 75 65 73 74 73 00 00 00 11 00 00 11 11 88 88 11 | uests           
0060 | 04 0F 44 69 73 63 6F 6E 6E 65 63 74 00 00 00 00 |   Disconnect    
0070 | 00 00 00 00 11 00 00 11 11 99 99 11 04 0F 43 6C |               Cl
0080 | 65 61 72 20 6C 69 63 65 6E 73 65 00 00 00 00 00 | ear license     
I 229195 2024-04-19 08:48:34 - [Commands] Received from C-1 (version=DC_V2 command=96 flag=00)
0000 | 96 00 0C 00 83 D8 34 8A 82 00 00 00             |       4         
I 229195 2024-04-19 08:48:34 - [Commands] Sending to C-1 (version=DC_V2 command=B1 flag=00)
0000 | B1 00 20 00 32 30 32 34 3A 30 34 3A 31 39 3A 20 |     2024:04:19: 
0010 | 30 38 3A 34 38 3A 33 34 2E 30 30 30 00 01 00 00 | 08:48:34.000    
I 229195 2024-04-19 08:48:34 - [Commands] Received from C-1 (version=DC_V2 command=99 flag=00)
0000 | 99 00 04 00                                     |                 
I 229195 2024-04-19 08:48:36 - [Commands] Received from C-1 (version=DC_V2 command=10 flag=00)
0000 | 10 00 0C 00 11 00 00 11 11 22 22 11             |          ""     
I 229195 2024-04-19 08:48:36 - [Commands] Sending to C-1 (version=DC_V2 command=97 flag=01)
0000 | 97 01 04 00                                     |                 
I 229195 2024-04-19 08:48:37 - [Commands] Received from C-1 (version=DC_V2 command=B1 flag=00)
0000 | B1 00 04 00                                     |                 
I 229195 2024-04-19 08:48:37 - [Commands] Sending to C-1 (version=DC_V2 command=B1 flag=00)
0000 | B1 00 20 00 32 30 32 34 3A 30 34 3A 31 39 3A 20 |     2024:04:19: 
0010 | 30 38 3A 34 38 3A 33 37 2E 30 30 30 00 01 00 00 | 08:48:37.000    
I 229195 2024-04-19 08:48:37 - [Commands] Received from C-1 (version=DC_V2 command=99 flag=00)
0000 | 99 00 04 00                                     |                 
W 229195 2024-04-19 08:48:37 - [Server] Error processing client command: unordered_map::at
I 229195 2024-04-19 08:48:37 - [Server] Client disconnected: C-1 on fd 55
I 229195 2024-04-19 08:48:37 - [C-1] Deleted
```

**Additional context**
I've obfuscated the access key in these logs, if you think it might be relevant, let me know and I'll get the information.

**Comment by fuzziqersoftware** [comment]
I suspected auto patching was at fault, but it doesn't seem to be the cause. It's hard to know what's really going on here without a stacktrace; would it be too much trouble to capture one? Ask the player to connect to a test server running under gdb with `"CatchHandlerExceptions": false,` in config.json, then when you see a breakpoint, run `bt` to get the trace.

**Comment by Matt-Swift** [comment]
I think it might be the auto patching - I removed them from the account file (which were ItemLossPrevention, RareDropNotifications, EnemyHPBars, DisableIdleDisconnect, and Decoction) and the player was able to connect again without issue.

I did try replicating the scenario on a test server under gdb, and we were unable to get it to happen again. This test server is using the new client-functions re-organisation though.

I'm gonna close this ticket and if it comes around again I'll get something more substantial by getting the logs from production.
